.github/workflows/release-management.yaml

```diff
@@ -0,0 +1,196 @@
+name: Release Management
+
+on:
+  release:
+    types:
+      - created
+
+env:
+  TAG_NAME: ${{ github.event.release.tag_name }}
+
+jobs:
+
+  generate_release_notes:
+    name: Generate Release Notes
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get previous tag
+        id: get_previous_tag
+        run: |
+          prevTag=$(git describe --abbrev=0 --tags $(git rev-list --tags --skip=1 --max-count=1))
+          echo "prev_tag=${prevTag}" >> $GITHUB_OUTPUT
+
+      - name: Generate release notes
+        env:
+          GITHUB_TOKEN: ${{ secrets.PROJECT_MANAGEMENT_TOKEN }}
+        run: |
+          ./scripts/changelog-script.sh ${{ steps.get_previous_tag.outputs.prev_tag }} ${TAG_NAME}
+
+      - name: Upload Release Notes Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: release-changelog
+          path: release-changelog.md
+          retention-days: 30
+
+  create_draft_release_pr:
+    name: Create Draft Release PR
+    runs-on: ubuntu-latest
+    needs: generate_release_notes
+    permissions:
+      pull-requests: write
+      contents: write
+
+    concurrency: release-pr-${{ github.event.release.tag_name }}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: main
+          fetch-depth: 0
+
+      - run: mkdir -p /tmp/release-notes
+      - uses: actions/download-artifact@v3
+        with:
+          name: release-changelog
+          path: /tmp/release-notes
+      - run: find /tmp/release-notes -name release-changelog.md
+
+      - name: Update VERSION file
+        run: echo ${TAG_NAME} > build/VERSION
+
+      - name: Create draft release blog article with release notes
+        id: generate_draft_blog_post
+        run: |
+          export blogFileName=docs/website/blog/$(date -I)-odo-${TAG_NAME}.md
+          cat <<EOF > "$blogFileName"
```

Comment:
Could we have this template placed in a separate file, so it is easier to update

Reply:
Done - see [`58d2a97` (#6403)](https://github.com/redhat-developer/odo/pull/6403/commits/58d2a97b9c844d0caa963df54a53b08e342cc606)